### PR TITLE
Docs page

### PR DESCRIPTION
**Pull request by majk-p**
Built with mdoc and docusaurus v2

You might want to have a look at following links when reviewing
- https://scalameta.org/mdoc/docs/why.html
- https://docusaurus.io/docs/deployment#triggering-deployment-with-github-actions

The `mdoc.yaml` is a hybrid of what both those documentations suggests, as `mdoc` is currently integrated with docusaurus v1.

This is not the final shape of this documentation. This PR aims to introduce the framework and workflow for documentation with minimal content. More detailed documentation should be introduced in next PRs.

### CODE REVIEW

**Comment by majk-p** [comment]
Published to https://ocadotechnology.github.io/sttp-oauth2/

**Comment by majk-p** [comment]
@kubukoz @tplaskowski please have a look

File: docs/authorization-code.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+---
+sidebar_position: 3
+description: Authroization code grant documentation
```
Comment (kubukoz) [inline docs/authorization-code.md:3]:
```suggestion
description: Authorization code grant documentation
```

File: docs/client-credentials.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+`ClientCredentials` and `ClientCredentialsProvider` expose methods that:
+- Obtain token via `requestToken`
+- `introspect` the token for it's details like `UserInfo`
+
+
+## `sttp-oauth2` backends
+
+- provide Client Credentials Backend, which is an interceptor for another backend and which can:
+  - fetch a token using ClientCredentialsProvider
+  - reuse the token multiple times using cache (default cache implementation may be overridden using appropriate constructor functions)
```
Comment (kubukoz) [inline docs/client-credentials.md:17]:
```suggestion
  - reuse the token multiple times using a cache (default cache implementation may be overridden using appropriate constructor functions)
```

File: docs/getting-started.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+To use this library add following dependency to your `build.sbt` file
+```scala
+"com.ocadotechnology" %% "sttp-oauth2" % "@VERSION@"
+```
+## Usage
+
+Depending on your use case, please see documentation for the grant you want to support.
+
+Each grant is implemented in an object with explicit return and error types on methods and additionally, TaglessFinal friendly `*Provider` interface.
```
Comment (kubukoz) [inline docs/getting-started.md:23]:
```suggestion
Each grant is implemented in an object with explicit return and error types on methods and additionally, Tagless Final friendly `*Provider` interface.
```

File: website/src/components/HomepageFeatures.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+import styles from './HomepageFeatures.module.css';
+
+const FeatureList = [
+  {
+    title: 'Built for OAuth2',
+    imagePath: "/sttp-oauth2/img/oauth2.png",
+    // Svg: require('../../static/img/undraw_docusaurus_mountain.svg').default,
+    description: (
+      <>
+        <code>sttp-oauth2</code> supports following following grant types: Authorization code, Password grant, Client credentials
```
Comment (kubukoz) [inline website/src/components/HomepageFeatures.js:12]:
```suggestion
        <code>sttp-oauth2</code> supports the following following grant types: Authorization code, Password grant, Client credentials
```

**Review by kubukoz — CHANGES_REQUESTED** [review]

**Review by kubukoz — APPROVED** [review]
lgtm

**Comment by kubukoz** [comment]
Move fast, break things :trollface:

File: .github/workflows/mdoc.yml [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+          if [ -e yarn.lock ]; then
+          yarn install --frozen-lockfile
+          elif [ -e package-lock.json ]; then
+          npm ci
+          else
+          npm i
+          fi
+          npm run build
+  gh-release:
+    if: github.event_name != 'pull_request'
```
Comment (kubukoz) [inline .github/workflows/mdoc.yml:29]: General wondering, will this run on pushes to non-main branches? Maybe it'd be better to run this only on main?

Comment (majk-p) [reply .github/workflows/mdoc.yml:29]: Currently it should publish on `main`, `docs` and tags. Before merging I'll drop `docs`.
```yaml
name: Website
on:
  push:
    branches: 
      - main
      - docs
    tags: ["*"]
```

Comment (kubukoz) [reply .github/workflows/mdoc.yml:29]: you mean before merging the next PR :D

**Pull request merged**